lib/libzfs/libzfs_import.c

```diff
@@ -820,13 +820,45 @@ pool_active(libzfs_handle_t *hdl, const char *name, uint64_t guid,
 	return (0);
 }
 
+boolean_t
+zfs_force_import_required(nvlist_t *config)
+{
+	uint64_t state;
+	uint64_t hostid = 0, orig_txg = 0;
+	unsigned long system_hostid = get_system_hostid();
+
+	state = fnvlist_lookup_uint64(config, ZPOOL_CONFIG_POOL_STATE);
+	(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_HOSTID,
+	    &hostid);
+	(void) nvlist_lookup_uint64(config, ZPOOL_CONFIG_IMPORT_TXG,
+	    &orig_txg);
+
+	/*
+	 * Logic in kernel excludes pools where spa_is_root == B_TRUE from the
+	 * force import requirement.  I cannot find how that is set.  Can that
+	 * be determined from the config nvlist?
+	 */
```

Comment:
This flag is set on the kernel side in `spa_import_rootpool()` and isn't available from user space.  This function was removed from ZoL in commit e5fe9ddeec55e3f7f8348330b3d414244f2d0670 since due to bootstrapping differences between Linux and Illumos it was all dead code.  In a separate change it would be worth revisiting getting this flag set correctly for Linux root pools.